### ISSUE TITLE
Compiled source fails to write to disk.

### ISSUE THREAD

**Issue by ryansobol** [brunch/postcss-brunch#17]
The current version of postcss-brunch v0.6.0 and the current version of brunch v2.8.2 fails to write the compiled source to disk.

See the [comment](https://github.com/iamvdo/postcss-brunch/issues/17#issuecomment-238769608) below.

**Comment by ryansobol** [comment]
Here's a working example of `postcss-brunch@0.6.0` failing to work with `brunch@2.8.2`.

https://github.com/ryansobol/postcss-brunch-bug

**Comment by ryansobol** [comment]
It took me all day, but I found the cause of the bug. 

The source map object, generated by `postcss` and returned by `postcss-brunch`, includes a `sources` property. In the linked project, the value of the `sources` property is `['index.css']`. Unfortunately, the actual relative path of this file is `app/styles/index.css`. I say unfortunately because Brunch is trying to read the `index.css` file from disk but failing.

https://github.com/ryansobol/postcss-brunch-bug/blob/master/node_modules/brunch/lib/fs_utils/source_file.js#L53

I have no idea what Brunch is attempting to do nor why it's doing it. I'll leave that to the contributors, @paulmillr, and @shvaikalesh to wrestle with. I wouldn't be surprised if there's a simple solution, but I don't have the expertise to figure it out in a timely fashion.

What's most concerning to me is threefold. First, why is the test suite for this plugin, and many other brunch plugins, so inadequate? Unit tests are a great first step, but what about integration tests? This bug renders the entire plugin dead in the water. How many other people have tried to use `postcss-brunch` and gave up when it didn't work? This plugin is extremely important to the Brunch community. PostCSS is the Babel of CSS. If it's appreciated, I'd be happy to add some integration tests once the plugin is working again.

Second, why do so many promises in Brunch have no `catch()` function? Brunch is a brilliant piece of software and congratulations are in order for closing over 1000 issues. But the lack of so many rejection handlers is makes it feel super unprofessional. Rather than displaying an error so I could report it, I spent all day learning Brunch's implementation and tracking down the cause of this bug.

https://github.com/ryansobol/postcss-brunch-bug/blob/master/node_modules/brunch/lib/fs_utils/source_file.js#L56

And third, why isn't Brunch's plugin API and pipeline for source maps better documented? A few diagrams and examples of the source map object would go a long way. Some code re-structuring of the internals wouldn't hurt either. It's pretty hard to follow the logic.

http://brunch.io/docs/plugins#reference

I'll update the title and description of this issue to better reflect this new information.

**Comment by paulmillr** [comment]
@ryansobol sorry you've run into the weird issue.

Absolutely agree with you. I just pushed the first promise rejection handler. We're going to add more handlers soon.

As for the integration tests, the reason for lack of tests if lack of resources. I developed about 15 Brunch plugins on my own & it's hard to maintain them even in the current situation.

Regarding plugin documentation, what do you think about the [Internals](http://brunch.io/docs/plugins#internals) section?

**Comment by KarlAmort** [comment]
I ran into the same problem and can confirm that it does not occur with postcss-brunch 0.5.0 / brunch 2.8.3.

Can also confirm that this ecosystem is even worse than webpack. Why anyone can consider it a good idea to have config files in 3 different syntaxes is beyond me, and of course they're all deeply nested and don't complain about mistakes. Each and every repository seems to be in some sort of major transition that makes them unable to say what it actually is that they do. Oh, and along the way someone renames the the configuration key "dependency" to "deps". 

(Sorry, 14 wasted with this, had to rant)

**Comment by paulmillr** [comment]
We want to take over this repository and make it official. This way it'll get better support. We also have some plans on incorporating postcss on a basic level.

@iamvdo would you be willing to transfer the rights to @brunch organization?

**Comment by paulmillr** [comment]
@iamvdo ping.

**Comment by iamvdo** [comment]
@paulmillr Sure, I'll transfer the rights right now.

Edit: I don't have admin rights to brunch org. How's the process to transfer this repo ?

**Comment by paulmillr** [comment]
@iamvdo I will add you there and give some rights. May need a few hours here.

**Comment by paulmillr** [comment]
@iamvdo added.

**Comment by paulmillr** [comment]
@iamvdo could you also add me to NPM?

`npm owner add paulmillr postcss-brunch`

**Comment by iamvdo** [comment]
@paulmillr Sure, done. ;)

**Comment by razor-x** [comment]
Version 2.0.1 seems to fix this for me, but only when using `.css` files. Sass files still aren't processed per #18.

**Comment by ryansobol** [comment]
@paulmillr, thank you so much for taking the time to improve this plugin. 😄
